### PR TITLE
fix(DataIterable): Update selection when items change.

### LINKED ISSUE #6094: [Bug Report] VDatatable: when an item is removed from the items list it should be removed from selection
### Versions and Environment
**Vuetify:** 1.4.1
**Vue:** 2.5.21
**Browsers:** Google Chrome, Mozilla Firefox
**OS:** Linux, Windows, Mac OSX

### Steps to reproduce
1. Select one or all items in the reproduction link.
2. Press the clear button

The datatable gets updated but not the selection.

### Expected Behavior
I would have expected the selection to be updated.

### Actual Behavior
The selection is not updated.

### Reproduction Link
<a href="https://codepen.io/anon/pen/qLMoMw" target="_blank">https://codepen.io/anon/pen/qLMoMw</a>

### PR DESCRIPTION
**Pull request by Nemikolh**
## Description
Update the selection when items has changed and some element where selected.

## Motivation and Context
Fixes #6094.

## How Has This Been Tested?
This has been tested visually, I'm also adding a unit tests (not done yet).

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-content>
      <v-container grid-list-xl>
        <v-data-table :items="items" v-model="selection">
          <template slot="items" slot-scope="props">
            <td><v-checkbox v-model="props.selected" primary hide-details></v-checkbox></td>
            <td>{{ props.item.id }}</td>
          </template>
        </v-data-table>
        {{selection}}
        <v-btn @click="clear">Clear</v-btn>
      </v-container>
    </v-content>
  </v-app>
</template>
<script>

export default {
  data: () => ({
     selection: [],
     items: [{ id: 'test' }, { id: 'foobar' }],
  }),
  methods: {
     clear() {
       this.items = [];
     }
  }
}
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] ~~New feature (non-breaking change which adds functionality)~~
- [ ] ~~Breaking change (fix or feature that would cause existing functionality to not work as expected)~~
- [ ] ~~Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)~~

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library) 
- [x] I've added a unit test

### CODE REVIEW

**Comment by Nemikolh** [comment]
Thanks! I'm working on it, should be done in a few minutes, hopefully.

**Comment by Nemikolh** [comment]
Took longer than expected, I had trouble debugging my test with Chrome until I found the `debug:test` script. Btw, is it normal that jest is missing from the list of devDependencies? I had to add it, to get the `debug:test` script working. (I haven't committed those changes though)

Let me know if I should change anything! :smiley:

**Comment by sh7dm** [comment]
@Nemikolh looks good, I'll approve it after checks pass!

File: packages/vuetify/src/mixins/data-iterable.js [hunk 1]
```diff
@@ -231,6 +231,29 @@ export default {
       if (this.pageStart >= this.itemsLength) {
         this.resetPagination()
       }
+      const itemsRemoved = {}
```
Comment (nekosaur) [inline packages/vuetify/src/mixins/data-iterable.js]: Alternative solution

```
const newItems = new Set(this.items.map(item => getObjectValueByPath(item, this.itemKey)))
const selection = this.value.slice().filter(item => newItems.has(getObjectValueByPath(item, this.itemKey)))

if (selection.length !== this.value.length) {
  this.$emit('input', selection)
}
```

Comment (Nemikolh) [reply packages/vuetify/src/mixins/data-iterable.js]: I didn't knew if using `Set` would be okay. It's much cleaner, I agree. Will push an update soon. Thanks for the review!

**Comment by Nemikolh** [comment]
Hi @sh7dm, @nekosaur, are there any other steps I need to take to unblock this?

**Comment by sh7dm** [comment]
@Nemikolh thanks!

**Pull request merged**